src/MongoDB.Driver.Core/Core/Operations/EndTransactionOperation.cs

```diff
@@ -242,7 +266,9 @@ private bool ShouldReplaceTransientTransactionErrorWithUnknownTransactionCommitR
                 return true;
             }
 
-            if (exception is MongoNotPrimaryException || exception is MongoNodeIsRecoveringException)
+            if (exception is MongoNotPrimaryException ||
+                exception is MongoNodeIsRecoveringException ||
+                exception is MongoExecutionTimeoutException) // MaxTimeMSExpired
```

Comment:
This line is about `top-level case` from here: https://github.com/mongodb/specifications/commit/8a2c678ff8fe9144870cec835554181dd6639fa3#diff-1c9bb9c2c4a17174d2c5d541152dd96bR271

    Note: a maxTimeMS error will have the MaxTimeMSExpired
    code (50) and can be reported as a top-level error ...

Reply:
`inside writeConcernError` is in line 290:

    default:
        return true;
    return true;